File: 590.n-ary-tree-postorder-traversal/solution.go
/**
 * Definition for a Node.
 * type Node struct {
 *     Val int
 *     Children []*Node
 * }
 */

func postorder(root *Node) []int {
    var res []int
    var dfs func(*Node)

    dfs = func(node *Node) {
        if node != nil {
            for _, child := range node.Children {
                dfs(child)
            }
            res = append(res, node.Val)
        }
    }

    dfs(root)
    return res
}

File: 590.n-ary-tree-postorder-traversal/solution.kt
/**
 * Definition for a Node.
 * class Node(var `val`: Int) {
 *     var children: List<Node?> = listOf()
 * }
 */

class Solution {
    fun postorder(root: Node?): List<Int> =
        root?.run { children.flatMap { postorder(it) } + `val` } ?: emptyList()
}